src/mpid/ch4/src/ch4r_rma.h

```diff
@@ -95,6 +95,10 @@ MPL_STATIC_INLINE_PREFIX int MPIDIG_do_put(const void *origin_addr, int origin_c
         MPIR_Datatype_get_true_lb(target_datatype, &am_hdr.target_true_lb);
         MPIR_T_PVAR_TIMER_END(RMA, rma_amhdr_set);
 
+        MPIDIG_send_init(sreq, origin_datatype, data_sz);
```

Comment:
If the transport is not going to do pipeline, then this `send_init` is a waste. I think we should do this `send_init` inside transport, that way, we only do the setup when we need it.

Reply:
that's fine.